src/bin/pg_autoctl/cli_common.c

```diff
@@ -266,16 +301,29 @@ int
 keeper_cli_getopt_pgdata(int argc, char **argv)
 {
 	KeeperConfig options = { 0 };
-	int c, option_index = 0;
+	int c, option_index = 0, errors = 0;
+	int verboseCount = 0;
 
 	static struct option long_options[] = {
```

Comment:
root_option() function and this function is now very similar.

Reply:
The difference is `--pgdata`. Sharing code in C for handling `switch` cases is... well... I could do some advanced macrology but I'd rather not... what do you think?

Reply:
This PR is not a good place for it but I have following thought.
1 - we provide long and short options for getopt_long() call. Both parameters need to be compatible and need to be changed together. Perhaps we could have a way to derive short options from long ones since we already specify corresponding short name when we are declaring long ones.

2 - Considering adding a group mask  to accompany `struct option` such that when we are declaring an option line we can also specify the group it belongs to. This way when parsing we also check the group mask. That could help handling similar parser functions. But this idea is pretty immature at this time. More thinking is needed.

Reply:
when I say group mask I think of a bitmask, so that the same option could be part of multiple groups.
```c
#define GETOPT_PGDATA 0x00001
#define GETOPT_CREATE_NODE 0x0002
```
so that when we are declaring an option we can say
```c
GETOPT_PGDATA | GETOPT_CREATE_NODE
```
Prior to running `getopt_long()` we can run a function to determine the options we are looking for 
with a group number.

Reply:
prefer functions over macros. Like the name `macrology`

Reply:
It might be preferable to switch to using https://www.gnu.org/software/libc/manual/html_node/Argp.html#Argp rather than continue using the getopt API in that case, but I didn't study that.